exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/AbstractParquetScanBatchCreator.java

```diff
@@ -191,4 +201,42 @@ protected AbstractDrillFileSystemManager(OperatorContext operatorContext) {
     protected abstract DrillFileSystem get(Configuration config, String path) throws ExecutionSetupException;
   }
 
+  /**
+   * Check whether any of queried columns is nested or repetitive.
+   *
+   * @param footer  Parquet file schema
+   * @param columns list of query SchemaPath objects
+   */
+  public static boolean containsComplexColumn(ParquetMetadata footer, List<SchemaPath> columns) {
+
+    Map<String, ColumnDescriptor> colDescMap = ParquetReaderUtility.getColNameToColumnDescriptorMapping(footer);
+    Map<String, SchemaElement> schemaElements = ParquetReaderUtility.getColNameToSchemaElementMapping(footer);
+
+    for (SchemaPath schemaPath : columns) {
+      // non-nested column check: full path must be equal to root segment path
+      if (!schemaPath.getUnIndexed().toString().replaceAll("`", "")
```

Comment:
That's definitely not a good choice of comparing by replacing "`". Please find a different way to compare or write helper method that would compare without replacing, maybe traversing schema path. Schema path has plenty of handy methods, if needed is missing implement one. Basically you cannot rely on to string representation it can change, you should rely on object info instead.

Reply:
Replaced with `SchemaPath.isLeaf()`